app/src/validators/object.js

```diff
@@ -12,11 +11,9 @@ const schema = {
     }),
     query: Joi.object((() => {
       const query = {
-        s3VersionId: Joi.string()
+        s3VersionId: Joi.string(),
```

Comment:
We don't need the IIFE for this anymore as it's now just a static object definition. This can be reduced to the following I believe:
```js
    query: Joi.object({
      s3VersionId: Joi.string(),
      versionId: type.uuidv4
    }).nand('s3VersionId', 'versionId')
```